catalog/transformer/catalog-transformer-pptx/src/test/java/ddf/catalog/transformer/input/pptx/PptxInputTransformerTest.java

```diff
@@ -0,0 +1,224 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.input.pptx;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.poi.openxml4j.util.Nullable;
+import org.apache.poi.xslf.usermodel.XMLSlideShow;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import ddf.catalog.transformer.input.tika.TikaInputTransformer;
+
+public class PptxInputTransformerTest {
+
+    private static List<String> files = Arrays.asList("/45541_Footer.pptx",
+            "/45541_Header.pptx",
+            "/45545_Comment.pptx",
+            "/49386-null_dates.pptx",
+            "/51187.pptx",
+            "/54542_cropped_bitmap.pptx",
+            "/56812.pptx",
+            "/KEY02.pptx",
+            "/OverlappingRelations.pptx",
+            "/SampleShow.pptx",
+            "/WithMaster.pptx",
+            "/ae.ac.uaeu.faculty_nafaachbili_GeomLec1.pptx",
+            "/alterman_security.pptx",
+            "/at.ecodesign.www_downloads_Vertiefungsvortrag_elektronik.pptx",
+            "/backgrounds.pptx",
+            "/layouts.pptx",
+            "/pie-chart.pptx",
+            "/pptx2svg.pptx",
+            "/prProps.pptx",
+            "/present1.pptx",
+            "/sample.pptx",
+            "/sample_pptx_grouping_issues.pptx",
+            "/shapes.pptx",
+            "/table_test.pptx",
+            "/testPPT.pptx",
+            "/themes.pptx",
+            "/with_japanese.pptx",
+            "/hello-world-signed.pptx",
+            "/hello-world-unsigned.pptx",
+            "/ms-office-2010-signed.pptx");
+
+    private final InputTransformer inputTransformer = new TikaInputTransformer(null);
+
+    private InputStream getResource(String resourceName) {
+        return PptxInputTransformerTest.class.getResourceAsStream(resourceName);
+    }
+
+    @Test(expected = CatalogTransformerException.class)
+    public void testBadCopy() throws IOException, CatalogTransformerException {
+        try {
+            PptxInputTransformer t = new PptxInputTransformer(inputTransformer);
+            InputStream is = Mockito.mock(InputStream.class);
+            Mockito.when(is.read(Mockito.any()))
+                    .thenThrow(new IOException());
+            t.transform(is);
+        } catch (CatalogTransformerException e) {
+            Assert.assertTrue(e.getMessage()
```

Comment:
OK, but I'm going to inject into mockito a specific instance of IOException and then make sure the cause is that instance. That way if the code under test through an IOException for some other reason, then the test will fail.